### DISCUSSION TITLE
Log task names as they execute

### DISCUSSION THREAD

**Discussion by mwisnicki** [sbt/sbt#7613] [Q&A]
Is it possible to print task names as they execute?

```scala
val bar = taskKey[Unit]("bar")
bar := {
  println("hello")
}
```

I tried `-v` and `--debug` but it didn't show me bar except in command line arguments.
`-Dsbt.task.timings=true` also didn't include the task.

https://github.com/sbt/sbt/issues/1209 is marked as closed yet I couldn't figure out what is the resolution. 
https://www.scala-sbt.org/release/docs/Howto-Logging.html does not seem very helpful unless the solution is to really write custom logger/plugin?

**Comment by eed3si9n** [comment]
I think the problem is that the `--timings` skips over the tasks that didn't take too long (1ms maybe?). With

```scala
val foo = taskKey[Unit]("foo")
val bar = taskKey[Unit]("bar")
foo := {
  Thread.sleep(1000)
  println("foo")
}
bar := {
  val _ = foo.value
  Thread.sleep(1000)
  println("bar")
}
```

```bash
$ sbt bar --timings
```

this shows:

```
[info] Total time: 2033 ms
[info]   foo / bar: 1016 ms
[info]   foo / foo: 1014 ms
```

This is also available with

```bash
$ sbt bar --traces
$ ls target/traces/build.trace
target/traces/build.trace
```

This can be loaded into Chrome `chrome://tracing/`

**Comment by mwisnicki** [comment]
It also doesn't print them as they execute but as summary after
